core/src/main/java/org/axonframework/common/IdentifierFactory.java

```diff
@@ -58,11 +57,18 @@
     }
 
     private static IdentifierFactory locateFactories(ClassLoader classLoader, String classLoaderName) {
+        logger.debug("Looking for IdentifierFactory implementation using the {} class loader", classLoaderName);
         IdentifierFactory found = null;
         Iterator<IdentifierFactory> services = ServiceLoader.load(IdentifierFactory.class, classLoader).iterator();
         if (services.hasNext()) {
-            logger.debug("Found IdentifierFactory implementation using the {} Class Loader", classLoaderName);
-            found = services.next();
+            try {
+                // here ServiceLoader verifies the correct type of the provided service class
+                found = services.next();
+                logger.debug("Found IdentifierFactory implementation using the {} class Loader", classLoaderName);
+            } catch (ServiceConfigurationError error) {
+                logger.warn("A service configuration error occurred for the provided IdentifierFactory: {}",
```

Comment:
I'm wondering whether we want to continue iterating through other services if the first one has issues with configuring. What's your take on that @jorgheymans ?

Reply:
true we could try the next one in that case if there is one. Different behavior though, if you think it's ok to do it that way i'll adapt the PR.

Reply:
I'm fine with it, maybe @abuijze or @smcvb have something against it?